### DISCUSSION TITLE
using putListingsItem to add offer to existing asin ?

### DISCUSSION THREAD

**Discussion by uk-2** [amzn/selling-partner-api-models#3214] [General]
@jevoniuk - Hi 
@prashanthseshadri 
@amz-tools
@IonicaBizau 

Use case :  Add offer to existing ASIN (price,qty,sku etc

(please correct  me if we are using the correct call)

i am trying to add a offer to existing listing using  putListingsItem , we have followed the documentation but we are unable to get any further
can you show us some example JSon where we add offer to existing asin 

LISTING_OFFER_ONLY

What do we need to use for product type ?

**Comment by tlleech** [comment]
Hi, have you found the solution?

**Comment by IonicaBizau** [comment]
Finally I chose to use the feeds api. It is a very complex and tedious
process...

**Comment by jashanbhullar** [comment]
You can use the [patchListingsItem](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference#patch-listings2021-08-01itemsselleridsku) operation to partially update the procject. 

```
{
  "productType": "PRPODUCT_TYPE",
  "patches": [
    {
      "op": "replace",
      "path": "/attributes/offer",
      "value": [
        {
          "value": "",
         .... more values
          "marketplace_id": "ATVPDKIKX0DER"
        }
      ]
    },
  }
 ```
 
 You can use [getDefinitionsProductType](https://developer-docs.amazon.com/sp-api/docs/product-type-definitions-api-v2020-09-01-reference#get-definitions2020-09-01producttypesproducttype) to find the exact values you need to send. Much easier than feeds API but still doesn't ensure that the updates will go through!

**Comment by vaidas-lungis** [comment]
productType `PRODUCT` ❓
